### ISSUE TITLE
Inconsistent Availability of Built-in cmdlets in Parallel Runspace

### ISSUE THREAD

**Issue by andyrobbins** [PowerShell/PowerShell#17150]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

I am experiencing a problem with runspaces created with ForEach-Object -Parallel. I believe each runspace should always have access to all default modules and cmdlets, but I am seeing inconsistencies where sometimes a runspace does not appear to have access to default modules and cmdlets.

The problem is reproducible when specifying a ThrottleLimit of 1, but is more reliably reproduced when specifying a ThrottleLimit greater than 1.

### Expected behavior

```console
Expected behavior, and the behavior that sometimes does occur:

PS /Users/andyrobbins> Get-Date

Wednesday, April 13, 2022 12:03:54 PM

PS /Users/andyrobbins> function GetDate {Get-Date}
PS /Users/andyrobbins> New-Variable -Name 'GetDateDefinition' -Value (Get-Content function:GetDate) -Force
PS /Users/andyrobbins> @(1..3) | ForEach-Object -ThrottleLimit 1 -Parallel { $function:GetDateThread = $using:GetDateDefinition; GetDateThread}

Wednesday, April 13, 2022 12:03:54 PM
Wednesday, April 13, 2022 12:03:54 PM
Wednesday, April 13, 2022 12:03:54 PM
```

### Actual behavior

```console
Actual behavior, which happens only some of the time:

PS /Users/andyrobbins> Get-Date

Wednesday, April 13, 2022 12:03:54 PM

PS /Users/andyrobbins> function GetDate {Get-Date}
PS /Users/andyrobbins> New-Variable -Name 'GetDateDefinition' -Value (Get-Content function:GetDate) -Force

PS /Users/andyrobbins> @(1..3) | ForEach-Object -ThrottleLimit 1 -Parallel { $function:GetDateThread = $using:GetDateDefinition; GetDateThread}

Wednesday, April 13, 2022 12:04:05 PM
Get-Date: The term 'Get-Date' is not recognized as a name of a cmdlet, function, script file, or executable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
Wednesday, April 13, 2022 12:04:05 PM
```

### Error details

```console
Get-Date: The term 'Get-Date' is not recognized as a name of a cmdlet, function, script file, or executable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
```

### Environment data

```powershell
PS /Users/andyrobbins> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.2.2
PSEdition                      Core
GitCommitId                    7.2.2
OS                             Darwin 21.1.0 Darwin Kernel Version 21.1.0: Wed Oct 13 17:33:01 PDT 2021; root:xnu-8019.41.5~1/RELEASE_ARM64_T6000
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

PS C:\Users\andyrobbins> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.2.1
PSEdition                      Core
GitCommitId                    7.2.1
OS                             Microsoft Windows 10.0.22000
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Visuals

<img width="1029" alt="Screen Shot 2022-04-13 at 12 07 31 PM" src="https://user-images.githubusercontent.com/842644/163253024-b8d99175-1452-4246-984c-114dcaa1543a.png">
<img width="1167" alt="Screen Shot 2022-04-13 at 12 07 57 PM" src="https://user-images.githubusercontent.com/842644/163253051-9d4358f0-f10d-4561-ace8-89e675217728.png">
<img width="1018" alt="Screen Shot 2022-04-13 at 12 11 24 PM" src="https://user-images.githubusercontent.com/842644/163253062-4fea5113-2e88-4af5-b3c9-e1430ac1da16.png">

**Comment by andyrobbins** [comment]
I believe this is a repeat of this issue: https://github.com/PowerShell/PowerShell/issues/16461

**Comment by SeeminglyScience** [comment]
Yeah this should throw since you're attempting to bring over a `ScriptBlock` as is.

The code that does this check currently may need to be updated to unwrap `PSObject`s. /cc @PaulHigin

**Comment by andyrobbins** [comment]
Thank you @SeeminglyScience.

I found that the workaround posted [here](https://github.com/PowerShell/PowerShell/issues/16461#issuecomment-967759037) by @daxian-dbw appears to fix the issue:

<img width="1407" alt="Screen Shot 2022-04-13 at 12 58 46 PM" src="https://user-images.githubusercontent.com/842644/163260324-22a0c515-693b-4933-96ba-c7bd89304f2b.png">

I collapsed the example multi-line script shared by @daxian-dbw into one line only to be able to fit more examples into the same screen space.

**Comment by SeeminglyScience** [comment]
Just to clarify, there is something potentially actionable here since it's evading the scriptblock check. This if:

https://github.com/PowerShell/PowerShell/blob/7cc9c874730e87b06dbe54d50e77c1c472996c54/src/System.Management.Automation/engine/InternalCommands.cs#L422-L431

Might need to be:

```csharp
if (item is ScriptBlock or PSObject { BaseObject: ScriptBlock })
{
    // etc
}
```

**Comment by PaulHigin** [comment]
@SeeminglyScience  Yes, but the point of this Issue is that the author expects ForEach-Object -Parallel to duplicate the original session into all parallel sessions, and this just isn't possible.  Adding a check for PSObject scriptblocks needs to be created in a separate issue.

**Comment by SeeminglyScience** [comment]
Ah good point Paul, I've opened #17152 to track. Thanks!
